src/components/Recovery.jsx

```diff
@@ -0,0 +1,178 @@
+// ============================================================================================
+// IMPORTS
+// ============================================================================================
+
+import { useContext } from "react";
+import PropTypes from "prop-types";
+import { UserContext } from "../contexts/UserContext.jsx";
+
+import { sendRequest } from "../scripts/sendrequest.js";
+import { getFormData } from "../scripts/getFormData.js";
+
```

Comment:
We can use validator.js to check for email format, password complexity, and matching password confirmation.

Reply:
The e-mail format should've already been validated at this point by the `Login.jsx` component.  However, an invalid e-mail address at this point would indicate the presence of a bug, so it additional validation could be done here.

*I need to add asserts & exceptions to the functions that deal with the stable parts of the database schema to make it easier to detect bugs.*

We haven't thought about password complexity yet.  We ~can~ should add that to our backlog.